src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
```

Comment:
Protobuf only uses char and does not use wchar_t. If GetModuleFileNameA is for compatibility, I think it's exactly what we should use here in protobuf. I agree with you new applications should probably consider not using it but in protobuf we are already manipulating every text as simple char arrays and there is no point to introduce wchar_t here.